File: app/src/main/java/tech/salroid/filmy/fragment/FullReadFragment.java
package tech.salroid.filmy.fragment;

import android.animation.Animator;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewAnimationUtils;
import android.view.ViewGroup;
import android.view.animation.DecelerateInterpolator;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import butterknife.BindView;
import butterknife.ButterKnife;
import tech.salroid.filmy.R;

/*
 * Filmy Application for Android
 * Copyright (c) 2016 Ramankit Singh (http://github.com/webianks).
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */


public class FullReadFragment extends Fragment implements View.OnClickListener {

    private String titleValue, descValue;

    @BindView(R.id.textViewTitle) TextView title;
    @BindView(R.id.textViewDesc) TextView desc ;
    @BindView(R.id.cross) ImageView crossButton ;


    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.read_full_layout, container, false);
        ButterKnife.bind(this,view);


        crossButton.setOnClickListener(this);


        // To run the animation as soon as the view is layout in the view hierarchy we add this
        // listener and remove it
        // as soon as it runs to prevent multiple animations if the view changes bounds
        view.addOnLayoutChangeListener(new View.OnLayoutChangeListener() {
            @Override
            public void onLayoutChange(View v, int left, int top, int right, int bottom, int oldLeft, int oldTop,
                                       int oldRight, int oldBottom) {
                v.removeOnLayoutChangeListener(this);

                int cx = getArguments().getInt("cx");
                int cy = getArguments().getInt("cy");

                // get the hypothenuse so the radius is from one corner to the other
                int radius = (int) Math.hypot(right, bottom);

                Animator reveal;

                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
                    reveal = ViewAnimationUtils.createCircularReveal(v, cx, cy, 0, radius);
                    reveal.setInterpolator(new DecelerateInterpolator(2f));
                    reveal.setDuration(1000);
                    reveal.start();
                }

            }
        });

        return view;
    }


    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (getArguments() != null) {
            titleValue = getArguments().getString("title", " ");
            descValue = getArguments().getString("desc", " ");
        }

    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        title.setText(titleValue);
        desc.setText(descValue);
    }

    @Override
    public void onClick(View view) {
        if (getFragmentManager() != null) {
            getFragmentManager().popBackStack();
        }

    }

}

File: app/src/main/java/tech/salroid/filmy/fragment/FullReadFragment.kt
package tech.salroid.filmy.fragment

import android.animation.Animator
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewAnimationUtils
import android.view.ViewGroup
import android.view.animation.DecelerateInterpolator
import androidx.fragment.app.Fragment
import tech.salroid.filmy.databinding.ReadFullLayoutBinding
import kotlin.math.hypot

class FullReadFragment : Fragment() {

    private var titleValue: String? = null
    private var descValue: String? = null

    private var _binding: ReadFullLayoutBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {

        _binding = ReadFullLayoutBinding.inflate(inflater, container, false)
        val view = binding.root

        binding.cross.setOnClickListener {
            if (fragmentManager != null) {
                fragmentManager?.popBackStack()
            }
        }

        view.addOnLayoutChangeListener(object : View.OnLayoutChangeListener {
            override fun onLayoutChange(
                v: View, left: Int, top: Int, right: Int, bottom: Int, oldLeft: Int, oldTop: Int,
                oldRight: Int, oldBottom: Int
            ) {
                v.removeOnLayoutChangeListener(this)
                val cx = arguments!!.getInt("cx")
                val cy = arguments!!.getInt("cy")

                val radius = hypot(right.toDouble(), bottom.toDouble())
                    .toInt()
                val reveal: Animator =
                    ViewAnimationUtils.createCircularReveal(v, cx, cy, 0f, radius.toFloat())

                reveal.interpolator = DecelerateInterpolator(2f)
                reveal.duration = 1000
                reveal.start()
            }
        })
        return view
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        titleValue = arguments?.getString("title", " ")
        descValue = arguments?.getString("desc", " ")
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.textViewTitle.text = titleValue
        binding.textViewDesc.text = descValue
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}